packages/contracts/src/2.0.0/protocol/Exchange/MixinAssetProxyDispatcher.sol

```diff
@@ -159,18 +159,22 @@ contract MixinAssetProxyDispatcher is
                 }
 
                 /////// Call `assetProxy.transferFrom` using the constructed calldata ///////
+                let inputLen := sub(cdEnd, cdStart)
                 let success := call(
                     gas,                    // forward all gas
                     assetProxy,             // call address of asset proxy
                     0,                      // don't send any ETH
                     cdStart,                // pointer to start of input
-                    sub(cdEnd, cdStart),    // length of input  
+                    inputLen,               // length of input  
                     cdStart,                // write output over input
                     512                     // reserve 512 bytes for output
                 )
                 if iszero(success) {
                     revert(cdStart, returndatasize())
                 }
+
+                // Increment free memory pointer
+                mstore(64, add(cdStart, inputLen))
```

Comment:
The memory is discarded within the asm block, so we don't need to increment the free memory pointer. (And if we do it should have been the max of `inputLen` and `returndatasize()` :) )

Note that Solidity does the same thing when it generates assembly for a call.

Gas cost will get *quadratically* worse for large batch orders.